File: 11-container-with-most-water/container-with-most-water.ts
function maxArea(height: number[]): number {
    let l = 0;
    let r = height.length - 1;
    let maxArea = 0;
    while (l < r) {
        if (height[l] < height[r]) {
            const area = height[l] * (r - l);
            maxArea = Math.max(area, maxArea);
            l++;
        }
        else {
            const area = height[r] * (r - l);
            maxArea = Math.max(area, maxArea);
            r--;
        }
    }
    return maxArea;
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int area = 0;
        while(left < right) {
            int curArea = (right - left) * Math.min(height[left] , height[right]);
            // System.out.printf("%d , %d , %d \n",left, right , curArea);
            area = Math.max(curArea , area);
            if(height[left] >= height[right]) right --;
            else left ++;
        }
        return area;
    }
}